tests/framework-tests/preferences/js/SelfVoicingEnactorTests.js

```diff
@@ -148,11 +129,21 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
      *******************************************************************************/
 
     fluid.defaults("fluid.tests.prefs.enactor.selfVoicingEnactor", {
-        gradeNames: ["fluid.prefs.enactor.selfVoicing", "fluid.tests.prefs.enactor.speakEnactor", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "fluid.tests.mock.textToSpeech", "fluid.prefs.enactor.selfVoicing", "autoInit"],
```

Comment:
Perhaps the need for this awkward grade list would be best met by refactoring fluid.prefs.enactor.selfVoicing so that it uses containment rather than grade merging to bind to the TTS impl. If it had a fluid.textToSpeech as a subcomponent rather than having it merged on to itself, it would be easier for this to be replaced, e.g. by distributeOptions or subcomponent overriding. The corruption of the grade structure emerged because it was ambiguous whether this component was intended to be a view component or not - the enactor is one, and the TTS isn't. This seems to be a clear sign that the relationship should be refactored.